### DISCUSSION TITLE
Error when signing: The specified item could not be found in the keychain

### DISCUSSION THREAD

**Discussion by raresn** [airsdk/Adobe-Runtime-Support#2881] [Q&A] [answered]
I am trying to sign my app on mac but i am failing
AdobeAIRSDK/bin/adt -sign -storetype KeychainStore -alias "Developer ID Application: On Time Agenda SRL" -target bundle xx.app xxx.app
And i get this error:
"error: The specified item could not be found in the keychain."

The key is clearly in the keychain.
![image](https://github.com/airsdk/Adobe-Runtime-Support/assets/1768694/6264d622-d20f-4d49-9d03-5c23ba9a47c8)

Anyone has any experience with macs or what i could do in this situation?

Thank you!

------
if i run:
security find-identity -p codesigning -v
-> 0 valid identities found

**Comment by raresn** [comment]
@ajwfrost maybe you have an idea?

**Reply by ajwfrost** [reply]
That error message, "The specified item could not be found", may actually be because it can't match the certificate name.. are you able to try with the full name (i.e. the team code in brackets at the end)? There may be differences between how forgiving the native codesign tool is, vs when we're using Java to extract the certificate from the keychain store programatically..

But I would also agree with pnico below, that you'd need to have a private key available in order to actually sign something. The certificate would generally appear under "My certificates" and have a button to be able to expand it and show the private key associated with it. You can perhaps import the private key using the Keychain Store app, I think it should then work out the association..? But I've never tried doing that separately...

thanks

**Comment by pnico** [comment]
do you have the private key that goes with the certificate in your keychain?

**Reply by raresn** [reply]
@pnico i do - was not generated on the mac but i think i have it. What do i do with it?

**Reply by pnico** [reply, marked as answer]
If you can get a .p12 file that has the key and the certificate included, usually just clicking on that on your mac will import both the certificate and the key to the keystore. if it works, you should see a caret to the left of the certificate icon in keychain assistant, which you can click to expand and then you'll see the key there with a little key icon (I don't see this in your image so this is probably what happened, you just have the certificate but not the private key)
We generate all our iOS certificates on macos so idk how this is done on windows, but you could try "Convert an Apple developer certificate to a P12 file on Windows" [here](https://help.adobe.com/en_US/air/build/WS5b3ccc516d4fbf351e63e3d118666ade46-7ff0.html#WSfffb011ac560372f46768d8712cd1d13954-7ffd) 
![Screen Shot 2023-10-20 at 9 32 50 AM](https://github.com/airsdk/Adobe-Runtime-Support/assets/680611/1b19dde1-0832-4d66-9575-826a8b5d8d82)

**Comment by raresn** [comment]
Importing a p12 worked!
Like you mentioned @pnico, after importing the p12 i now i have the expand option on the certificate!
Thank you and @ajwfrost!
I am off to get new errors now :))
